sdk/mcp-client-manager/index.ts

```diff
@@ -494,6 +500,64 @@ export class MCPClientManager {
     }
   }
 
+  async getResources(serverIds?: string[]): Promise<MCPResource[]> {
+    const targetServerIds = serverIds ?? this.listConnectedServers();
+    const allResources: MCPResource[] = [];
+
+    for (const serverId of targetServerIds) {
+      const client = this.getClient(serverId);
+      if (client) {
+        try {
+          const result = await client.listResources({});
+          if (result && Array.isArray(result.resources)) {
+            const resourcesWithServerId = result.resources.map((r: any) => ({
+              ...r,
+              _serverId: serverId,
+            }));
+            allResources.push(...resourcesWithServerId);
+          }
+        } catch (error) {
+          console.error(
+            `[MCPClientManager] Failed to list resources for server ${serverId}:`,
+            error,
+          );
+        }
+      }
+    }
+    return allResources;
+  }
+
+  async getResourceTemplates(
+    serverIds?: string[],
+  ): Promise<MCPResourceTemplate[]> {
+    const targetServerIds = serverIds ?? this.listConnectedServers();
+    const allResourceTemplates: MCPResourceTemplate[] = [];
+
+    for (const serverId of targetServerIds) {
+      const client = this.getClient(serverId);
+      if (client) {
+        try {
+          const result = await client.listResourceTemplates({});
+          if (result && Array.isArray(result.resourceTemplates)) {
+            const templatesWithServerId = result.resourceTemplates.map(
+              (t: any) => ({
+                ...t,
+                _serverId: serverId,
+              }),
+            );
+            allResourceTemplates.push(...templatesWithServerId);
+          }
+        } catch (error) {
+          console.error(
+            `[MCPClientManager] Failed to list resource templates for server ${serverId}:`,
+            error,
+          );
+        }
+      }
+    }
+    return allResourceTemplates;
+  }
+
```

Comment:
this is merely a prototype for now I still want to improve the usage but at least it allows me to test resources